src/Altinn.Platform/Altinn.Platform.Storage/DataCleanup/NightlyCleanup.cs

```diff
@@ -60,14 +64,22 @@ public async Task Run([TimerTrigger("0 0 3 * * *", RunOnStartup = false)] TimerI
                     if (dataElementsDeleted)
                     {
                         dataElementMetadataDeleted = await _cosmosService.DeleteDataElementDocuments(instance.Id);
+                        dataElementsBackupDeleted = await _blobService.DeleteDataBackup(instance.Id);
                     }
 
                     if (autoDeleteAppIds.Contains(instance.AppId))
                     {
                         instanceEventsDeleted = await _cosmosService.DeleteInstanceEventDocuments(instance.Id, instance.InstanceOwner.PartyId);
+                        instanceEventsBackupDeleted = await _blobService.DeleteInstanceEventsBackup(instance.Id, instance.InstanceOwner.PartyId);
                     }
 
-                    if (dataElementMetadataDeleted && (!autoDeleteAppIds.Contains(instance.AppId) || instanceEventsDeleted))
+                    instanceBackupDeleted = await _blobService.DeleteInstanceBackup(instance.Id, instance.InstanceOwner.PartyId);
```

Comment:
a bit unsure of the placement of the sanation of the instance backup. 

If we do it after metadata in cosmos is deleted, we have no way of re-trying if deletion of backup fails..
When using current placement we risk deleting the backup before the metadat from cosmos is deleted